src/MuTalk-Tests/MutationTestingPercentageOfMutantsBudgetTest.class.st

```diff
@@ -0,0 +1,62 @@
+Class {
+	#name : 'MutationTestingPercentageOfMutantsBudgetTest',
+	#superclass : 'MutationTestingBudgetTest',
+	#instVars : [
+		'numberOfMutations'
+	],
+	#category : 'MuTalk-Tests',
+	#package : 'MuTalk-Tests'
+}
+
+{ #category : 'running' }
+MutationTestingPercentageOfMutantsBudgetTest >> runAnalysisFor: aPercentage [
+
+	| result |
+	result := self
+		          runAnalysisWithBudget:
+		          (MutationTestingPercentageOfMutantsBudget for: aPercentage)
+		          on: { AuxiliarClassForMutationTestingAnalysis }
+		          withTests: { AuxiliarClassForMutationTestingAnalysisTest }.
+	numberOfMutations := analysis mutations size.
+	^ result
+]
+
+{ #category : 'tests' }
+MutationTestingPercentageOfMutantsBudgetTest >> testEvaluateAllMutantsWithMoreThanOneHundredPercentConstraint [
+
+	self runAnalysisFor: 120.
+	self
+		assert:
+			analysis generalResult numberOfEvaluatedMutants / numberOfMutations * 100
+		equals: 100
+]
+
+{ #category : 'tests' }
+MutationTestingPercentageOfMutantsBudgetTest >> testEvaluateTheCorrectPercentageOfMutants [
+
+	self runAnalysisFor: 20.
+	self
+		assert:
+			analysis generalResult numberOfEvaluatedMutants / numberOfMutations * 100
```

Comment:
This works because we have a _round_ number of mutations right?

Because if we had 33 mutations, the 20% of 33 is 6,6 mutations.
I imagine we don't execute partially mutations. So, what do we do in that case? 6 mutations or 7 mutations? :D

Reply:
Actually, my point is also that if we change the number of mutations the test will break.
Take my example of 33 mutations, and let's suppose we chose 20% rounded up, running 7 mutations (6 or 7 here is annecdotic).

Then the assertion 

`analysis generalResult numberOfEvaluatedMutants / numberOfMutations * 100`

Will be equals to `7/33*100 == 21.21` which is not your expected 20 :)

Equally, if we choose to run 6 instead of 7, we would have `6/33*100 == 18.18`.

At the end the question is: do we stop before exceeding the budget or after?
So we need a test for this ^^